File: Frontend/src/schemas/transacao.ts
import { z } from 'zod'

export const transacaoSchema = z.object({
  descricao: z
    .string()
    .min(1, 'Descrição é obrigatória')
    .max(400, 'Descrição deve ter no máximo 400 caracteres'),
  valor: z
    .number({ invalid_type_error: 'Valor deve ser um número' })
    .positive('Valor deve ser maior que 0'),
  tipo: z.coerce.number().int().min(0).max(1),
  categoriaId: z.string().uuid('Selecione uma categoria'),
  pessoaId: z.string().uuid('Selecione uma pessoa'),
})

export type TransacaoFormData = z.infer<typeof transacaoSchema>

File: Backend/src/ResiGa.Bkd.Domain/Models/Transacao.cs/Transacao.cs
namespace ResiGa.Bkd.Domain.Models.Transacao;

public class Transacao
{
    public Guid? Id { get; set; }
    public string Descricao { get; set; } = "";
    public decimal Valor { get; set; }
    public int Tipo { get; set; }
    public Guid CategoriaId { get; set; }
    public Guid PessoaId { get; set; }
}
